content/docs/rfcs/73/README.md

```diff
@@ -0,0 +1,199 @@
+Reputation may also play a role.
+For Waku light protocols, we envision a combination of monetary and reputation-based incentivization.
+See a [write-up on incentivization](https://github.com/waku-org/research/blob/master/incentivization.md) for our high-level reasoning on the topic.
+
+# Theory / Semantics
+
+Consider a light protocol with two roles: a client and a server.
+A server MAY indicate to a client that it expects certain eligibility criteria to be met.
+In that case, a client MUST provide a valid eligibility proof as part of its request.
+
+Forms of eligibility proofs include:
+- Proof of payment: for paid non-authenticated requests. A proof of payment, in turn, may also take different forms, such as a transaction hash or a ZK-proof. In order to interpret a proof of payment, the server needs information about its type.
+- Proof of membership: for services for a predefined group of users. An example use case: an application developer pays in bulk for their users' requests. A client then submits a proof of membership in the user set of that application. A similar concept (RLN) is used in Waku Relay for spam prevention.
+- Service credential: a proof of membership in a set of clients who have prepaid for the service (which may be considered a special case of proof of membership).
+
+Upon a receiving a request:
+- the server MUST check if the eligibility proof is included and valid;
+- if that proof is absent or invalid, the server MUST send back response with a corresponding error code and error description;
+- if the proof is valid, the server MUST send back the response that the client has requested.
+
+# Wire Format Specification / Syntax
+
+A client includes an `EligibilityProof` in its request.
+A server includes an `EligibilityStatus` in its response.
+
+```protobuf
+syntax = "proto3";
+
+message EligibilityProof {
+	optional bytes proof_of_payment = 1;  // e.g., a txid
+	optional bytes proof_of_membership = 2;  // e.g., an RLN proof
+	optional bytes service_credential = 3;
+	optional bytes other_eligibility_proof = 4;
+}
+
+message EligibilityStatus {
+  optional uint32 status_code = 1;
+  optional string status_desc = 2;
+}
+```
+
+We include the `other_eligibility_proof` field in `EligibilityProof` to reflect other types of eligibility proofs that could be added to the protocol later.
+
+# Implementation in Store (PoC version)
+
+This Section describes a proof-of-concept (PoC) implementation of incentivization in the Store protocol.
+Note: this section may later be moved to Store RFC.
+
+Store is one of Waku's light (i.e., client-server) protocols.
+A Store client queries the server for historic messages.
+A Store server responds with a list of messages that pass the user's filter.
+See [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) for the definitions of `HistoryQuery` and `HistoryResponse`.
+
+The PoC Store incentivization makes the following simplifying assumptions:
+- the client knows the server's on-chain address `A`;
```

Comment:
is this address hardcoded in some way? Perhaps it should be part of the req/resp interaction, otherwise it wouldn't be practical?

Reply:
Fair point!

The issue here is that it's the client who initiates the communication. Therefore, for the server to share its address, we'd need two rounds:

- client: I want this piece of data
- server: here is my address for payment
- client: here is the proof of payment
- server: here is the data

For this initial version, I prioritized simplicity. I asked a related question [here](https://github.com/waku-org/research/issues/35#issuecomment-1825572739) on how to balance simplicity for PoC (i.e., being just one-round) with practicality (negotiating prices prior to actual payment-data exchange). We've also discussed this on [2023-11-27 research sync](https://www.notion.so/Waku-Research-Meeting-Notes-837e169b7e3047679dfc62f8ed44294f#089d3a674d7441608eaa1c7713c02256) and came to the conclusion to make the PoC as simple as possible (i.e., one-round) and iterate from there.

We'll likely add a second round later (to negotiate prices, etc), it's just a question of priority to get something simple implemented first.